sharding/mainchain/smc_client.go

```diff
@@ -168,17 +169,29 @@ func (s *SMCClient) SMCFilterer() *contracts.SMCFilterer {
 	return &s.smc.SMCFilterer
 }
 
-// TransactionReceipt allows an SMCClient to retrieve transaction receipts on
-// the mainchain by hash.
-func (s *SMCClient) TransactionReceipt(hash common.Hash) (*types.Receipt, error) {
+// WaitForTransaction waits for transaction to be mined and returns an error if it takes
+// too long
+func (s *SMCClient) WaitForTransaction(hash common.Hash, durationInSeconds int64) error {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Duration(durationInSeconds)*time.Second)
```

Comment:
Rather than using context.Background as the parent context, could you please require context as an argument? We are using contexts to cancel processes and this long time out might need to be canceled early.

Providing the parent context from the function arguments would solve that issue

Reply:
In fact, you could have the caller provide a context from the beginning and they can set the timeout, if any.

Reply:
Also I don’t think you are using this context timeout properly. You never call ctx.Done()